src/xcode/ENA/ENA/Source/Client/HTTP Client/URLSession+Convenience.swift

```diff
@@ -101,15 +111,19 @@ extension URLSession.Response {
 	enum Failure: Error {
 		/// The session received an `Error`.
 		case httpError(Error, HTTPURLResponse)
-		/// The session did not receive an error but nor either an `HTTPURLResponse`/HTTP body.
-		case noResponse
 		case teleTanAlreadyUsed
 		case qrAlreadyUsed
 		case qrDoesNotExist
 		case regTokenNotExist
 		case invalidResponse
+		case noNetworkConnection
 		case serverError(Int)
 		case fakeResponse
+
+		/// **[Deprecated]** Legacy state to indicate no (meaningful) response was given.
+		///
+		/// This had multiple reasons (offline, invalid payload, etc.) and was ambiguous. Please consider another state (e.g. `invalidResponse` or `noNetworkConnection`) and refactor existing solutions!
+		case noResponse
```

Comment:
I'm fine with deprecation.
just a note: noResponse gets used inside the WiFiClient as a 'result' to get the loading done. This needs a refactoring if it gets removed later.